migrations/sqls/20220111134902-charge-purposes-up.sql

```diff
@@ -0,0 +1,31 @@
+
+/* Replace with your SQL commands */
+CREATE TABLE IF NOT EXISTS "water"."charge_purposes" (
+    "charge_purpose_id" uuid NOT NULL DEFAULT public.gen_random_uuid(),
+    "charge_element_id" uuid NOT NULL,
+    "abstraction_period_start_day" smallint NOT NULL,
+    "abstraction_period_start_month" smallint NOT NULL,
+    "abstraction_period_end_day" smallint NOT NULL,
+    "abstraction_period_end_month" smallint NOT NULL,
+    "authorised_annual_quantity" numeric NOT NULL,
+    "season" water.charge_element_season NOT NULL,
+    "season_derived" water.charge_element_season NOT NULL,
+    "source" water.charge_element_source NOT NULL,
+    "loss" water.charge_element_loss NOT NULL,
+    "factors_overridden" boolean NOT NULL,
+    "billable_annual_quantity" numeric,
+    "time_limited_start_date" date,
+    "time_limited_end_date" date,
+    "description" varchar,
+    "date_created" timestamp NOT NULL DEFAULT NOW(),
+    "date_updated" timestamp NOT NULL DEFAULT NOW(),
+    "is_test" boolean DEFAULT false NOT NULL,
+    "purpose_primary_id" uuid not null,
```

Comment:
purpose_primary_id, purpose_secondary_id and purpose_use_id should be foreign keys/constained